app/ui/legacy/src/main/java/com/fsck/k9/ui/messageview/RecipientNamesView.kt

```diff
@@ -0,0 +1,184 @@
+
+private const val MAX_NUMBER_OF_RECIPIENT_NAMES = 5
+
+/**
+ * View that displays the names of recipients of a message.
+ *
+ * Up to [MAX_NUMBER_OF_RECIPIENT_NAMES] names of recipients are displayed, followed by the number of recipients that
+ * weren't displayed.
+ *
+ * Examples:
+ * - to me, Alice, Bob, Charly +3
+ * - to Camila Hyphenated-Nam… +5
+ *
+ * This custom layout uses [RecipientLayoutCreator] to figure out how many recipient names can be displayed without
+ * being truncated. If not even one recipient name can be displayed without being truncated, we first measure the space
+ * needed for number of additional recipients, then use the rest to display the first recipient and ellipsize the end.
+ */
+class RecipientNamesView(context: Context, attrs: AttributeSet?) : ViewGroup(context, attrs) {
+    val maxNumberOfRecipientNames: Int = MAX_NUMBER_OF_RECIPIENT_NAMES
+
+    private val recipientLayoutCreator: RecipientLayoutCreator
+
+    private val recipientNameTextView: TextView
+    private val recipientCountTextView: TextView
+    private val additionRecipientSpacing: Int
+
+    init {
+        LayoutInflater.from(context).inflate(R.layout.recipient_names, this, true)
+        recipientNameTextView = findViewById(R.id.recipient_names)
+        recipientCountTextView = findViewById(R.id.recipient_count)
+        additionRecipientSpacing = (recipientCountTextView.layoutParams as MarginLayoutParams).marginStart
+    }
+
+    private var recipientNames: List<CharSequence> = emptyList()
+    private var numberOfRecipients: Int = 0
+
+    private val textMeasure = object : TextMeasure {
+        override fun measureRecipientNames(text: CharSequence): Int {
+            return measureWidth(recipientNameTextView, text)
+        }
+
+        override fun measureRecipientCount(text: CharSequence): Int {
+            return measureWidth(recipientCountTextView, text)
+        }
+
+        private fun measureWidth(textView: TextView, text: CharSequence): Int {
+            textView.text = text
+
+            val widthMeasureSpec = MeasureSpec.makeMeasureSpec(0, MeasureSpec.UNSPECIFIED)
+            val heightMeasureSpec = MeasureSpec.makeMeasureSpec(height, MeasureSpec.AT_MOST)
+            textView.measure(widthMeasureSpec, heightMeasureSpec)
+
+            return textView.measuredWidth
+        }
+    }
+
+    init {
+        recipientLayoutCreator = RecipientLayoutCreator(
+            textMeasure = textMeasure,
+            maxNumberOfRecipientNames = MAX_NUMBER_OF_RECIPIENT_NAMES,
+            recipientsPrefix = context.getString(R.string.message_view_recipient_prefix),
+            additionalRecipientSpacing = additionRecipientSpacing,
+            additionalRecipientsPrefix = context.getString(R.string.message_view_additional_recipient_prefix)
+        )
+
+        if (isInEditMode) {
+            recipientNames = listOf(
+                "Grace Hopper", "Katherine Johnson", "Margaret Hamilton", "Adele Goldberg", "Steve Shirley"
```

Comment:
Should this be here? Feels like a leftover from testing

Reply:
Fixed by c2eb4ff0a2f1889142be880b2e0183d9d19ffde6

Reply:
To actually answer your question: You're right, this code shouldn't really be there. But I think it's the least bad option. The preview in Android Studio was super useful in writing this custom view. And this was the easiest way I could think of to display test data. I believe having the test data in there will be valuable for developers looking at this class in the future.